src/osd/PrimaryLogPG.cc

```diff
@@ -6914,6 +6914,8 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 	  if (need_inc_ref) {
 	    result = -EINPROGRESS;
 	    break;
+	  } else {
+	    delete fin;
```

Comment:
Oh, I see this fix now.

This is needlessly difficult to follow -- a small change to inc_refcount_by_set or refcount_manifest could result in undefined behavior here if the reader doesn't notice the (undocumented) coupling between the return value of inc_refcount_by_set and the lifetime of cb.

Move the allocation of RefCountCallback into the block in inc_refcount_by_set where it is deterministically passed to refcount_object (you'll have to pass osd_op into inc_refcount_by_set).